code/modules/vehicles/tank.dm

```diff
@@ -0,0 +1,381 @@
+	if(!position)
+		return
+	if(pilot && position == "pilot")
+		to_chat(user, "[pilot] is already controlling [src]!")
+		return
+	if(gunner && position == "gunner")
+		to_chat(user, "[gunner] is already controlling [src]!")
+		return
+	if(can_enter(user)) //OWO can they enter us????
+		to_chat(user, "You climb into [src] as a [position]!")
+		return enter(user, position) //Yeah i could do this with a define, but this way we're not using multiple things
+
+/obj/vehicle/tank/proc/can_enter(var/mob/living/carbon/M) //NO BENOS ALLOWED
+	if(!M.IsAdvancedToolUser())
+		to_chat(M, "<span class='warning'>You don't have the dexterity to drive [src]!</span>")
+		return FALSE
+	if(!allowed(M))
+		to_chat(M, "<span class='warning'>Access denied.</span>")
+		return FALSE
+	var/obj/item/offhand = M.get_inactive_held_item()
+	if(offhand && !(offhand.flags_item & (NODROP|DELONDROP)))
+		to_chat(M, "<span class='warning'>You need your hands free to climb on [src].</span>")
+		return FALSE
+
+	if(!M.mind || !(!M.mind.cm_skills || M.mind.cm_skills.large_vehicle >= SKILL_LARGE_VEHICLE_TRAINED))
+		M.visible_message("<span class='notice'>[M] fumbles around figuring out how to get into the [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to get into [src].</span>")
+	var/time = 10 SECONDS - (2 SECONDS * M.mind.cm_skills.large_vehicle)
+	if(do_after(M, time, TRUE, src, BUSY_ICON_BUILD))
+		return TRUE
+
+/obj/vehicle/tank/proc/enter(var/mob/user, var/position) //By this point, we've checked that the seats are actually empty, so we won't need to do that again HOPEFULLY
+	if(!user || !position)
+		return //Something fucked up. Whoops!
+	user.forceMove(src)
+	operators += user
+	switch(position)
+		if("pilot")
+			pilot = user
+		if("gunner")
+			gunner = user
+
+
+/obj/vehicle/tank/proc/remove_all_players()
+	for(var/M in operators)
+		exit_tank(M)
+
+/obj/vehicle/tank/proc/exit_tank(var/mob/user) //By this point, we've checked that the seats are actually empty, so we won't need to do that again HOPEFULLY
+	if(!user)
+		return //Something fucked up. Whoops!
+	if(user == pilot)
+		user.forceMove(get_turf(src))
+		pilot = null
+		operators -= user
+	if(user == gunner)
+		user.forceMove(get_turf(src))
+		gunner = null
+		operators -= user
+
+/obj/vehicle/tank/relaymove(mob/user, direction)
+	if(user.incapacitated() || user != pilot)
+		to_chat(user, "You can't reach the gas pedal from down here, maybe try manning the driver's seat?")
+		return
+	if(world.time > last_move_time + move_delay)
+		. = step(src, direction)
+	update_icon()
+
+/obj/vehicle/tank/Bump(atom/A, yes)
+	. = ..()
+	var/facing = get_dir(src, A)
+	var/turf/temp = get_turf(loc)
+	var/turf/T = get_turf(loc)
+	A.tank_collision(src, facing, T, temp)
+	if(isliving(A) && pilot)
+		log_attack("[pilot] drove over [A] with [src]")
+
+/client/MouseDown(object, location, control, params)
+	. = ..()
+	if(istype(mob.loc, /obj/vehicle/tank)) //mob is a var that exists on all clients anyway
```

Comment:
This could be sending a signal instead, which would be registered by the tank once the mob enters the gunner seat, and unregister when it leaves it.
Let's see what @ninjanomnom says about it.

Also, there's no need to redefine it here. Add that coded in `code\_onclick\drag_drop.dm` where it's defined, instead of spreading little bits on several files.